3scaleAdapter/pkg/threescale/proxy_conf.go

```diff
@@ -80,15 +86,31 @@ func (pc *ProxyConfigCache) RefreshCache() {
 	log.Debugf("refreshing cache for existing proxy config entries")
 	forRefresh := pc.markForRefresh()
 	for _, store := range forRefresh {
+
 		cacheKey := pc.getCacheKeyFromCfg(store.replayWith.cfg)
+		host := pc.splitHostFromCacheKey(cacheKey)
+		if pc.isMisbehaving(host) {
+			continue
+		}
+
 		pce, err := getFromRemote(store.replayWith.cfg, store.replayWith.client, pc.metricsReporter.ReportMetrics)
 		if err != nil {
 			log.Infof("error fetching from remote while refreshing cache for service id %s", store.replayWith.cfg.ServiceId)
-			pc.delete(cacheKey)
-			break
+			pc.addMisbehavingHost(host, err)
+			continue
 		}
 		pc.set(cacheKey, pce, store.replayWith)
 	}
+
+	// this will cause failed hosts to have a chance to refresh their cached
+	// config a single time before they will be purged in the next loop
+	if len(pc.misbehavingHosts) > 0 {
+		pc.misbehavingHosts = make(map[string]bool)
+		go func() {
+			pc.refreshCacheFuture((pc.ttl - pc.refreshBuffer) / 2)
```

Comment:
@unleashed perhaps making this `2` configurable t determine the amount of retries within that timeframe is worthwhile?

Reply:
ah, didn't see this comment, good :+1: